pkg/ast/spl/spl.peg

```diff
@@ -4365,14 +4415,56 @@ Offset <- off:(RelIntegerAsString) tuOff:(RelTimeUnit) {
     return relTimeOffset, nil
 }
 
+ChainedRelativeTimestamp <- first:(Offset / Snap) rest:((Offset / Snap)*) {
+    var offsets []ast.RelativeTimeOffset
+    var snaps []string
+
+    // Append zero values to ensure corresponding offsets and snaps
+    // This ensures that each snap and offset correspond to the same index
+    appendZeroValues := func() {
+        if len(offsets) > len(snaps) {
+            snaps = append(snaps, "")
+        } else if len(snaps) > len(offsets) {
+            offsets = append(offsets, ast.RelativeTimeOffset{})
+        }
+    }
+
+    // Process the first element
+    if first != nil {
+        switch v := first.(type) {
+        case ast.RelativeTimeOffset:
+            offsets = append(offsets, v)
+            snaps = append(snaps, "") // Append zero value for Snap
+        case string:
+            snaps = append(snaps, v)
+            offsets = append(offsets, ast.RelativeTimeOffset{}) // Append zero value for Offset
+        }
```

Comment:
Also change this comment, if you change the variable name. As assigning empty `string` or `ast.RelativeTimeOffset` and the comment indicating `Zero` values does not make sense.

Reply:
Addressed it in the latest commit.